### ISSUE TITLE
tensorflow's Filling shuffle buffer is slow to fill up on very large datasets

### ISSUE THREAD

**Issue by idroz** [beringresearch/ivis#56]
Before each epoch, tensorflow fills up a shuffle buffer:

`Filling up shuffle buffer (this may take a while): 69 of 79`

This is not optimal behaviour for large datasets. Potential solution here: https://github.com/tensorflow/tensorflow/issues/30646#issuecomment-511359604

**Comment by Szubie** [comment]
This behaviour appears to be caused by the way TensorFlow 2.0 deals with keras Sequence objects.

The issue appears to be fixed in TensorFlow 2.1.0 rc2 (https://github.com/tensorflow/tensorflow/releases/tag/v2.1.0-rc2), so upgrading to that using `pip install --pre tensorflow==2.1.0-rc2` should fix the issue. Downgrading to TensorFlow 1.14 also fixes the issue.

Since this will remain a problem until TensorFlow 2.1.0 rc2 is officially released, I will leave this issue open until then.

**Comment by idroz** [comment]
TF 2.1 has been released and the issue seems to have resolved. Closing for now.
